File: bootstrap/archived_runa_0.0.7.5+/v0.0.8.4/benchmarks/rust/nqueens.rs
fn is_safe(board: &Vec<Vec<i32>>, row: usize, col: usize, n: usize) -> bool {
    for i in 0..col {
        if board[row][i] == 1 {
            return false;
        }
    }

    let mut i = row as i32 - 1;
    let mut j = col as i32 - 1;
    while i >= 0 && j >= 0 {
        if board[i as usize][j as usize] == 1 {
            return false;
        }
        i -= 1;
        j -= 1;
    }

    let mut i = row + 1;
    let mut j = col as i32 - 1;
    while i < n && j >= 0 {
        if board[i][j as usize] == 1 {
            return false;
        }
        i += 1;
        j -= 1;
    }

    true
}

fn solve(board: &mut Vec<Vec<i32>>, col: usize, n: usize) -> bool {
    if col >= n {
        return true;
    }

    for i in 0..n {
        if is_safe(board, i, col, n) {
            board[i][col] = 1;
            if solve(board, col + 1, n) {
                return true;
            }
            board[i][col] = 0;
        }
    }

    false
}

fn main() {
    let n = 8;
    let mut board = vec![vec![0; n]; n];
    solve(&mut board, 0, n);
    std::process::exit(0);
}

File: bootstrap/archived_runa_0.0.7.5+/v0.0.8.4.5/benchmarks/c/nqueens.c
#include <stdlib.h>

int is_safe(int** board, int row, int col, int n) {
    for (int i = 0; i < col; i++) {
        if (board[row][i]) return 0;
    }

    for (int i = row - 1, j = col - 1; i >= 0 && j >= 0; i--, j--) {
        if (board[i][j]) return 0;
    }

    for (int i = row + 1, j = col - 1; i < n && j >= 0; i++, j--) {
        if (board[i][j]) return 0;
    }

    return 1;
}

int solve(int** board, int col, int n) {
    if (col >= n) return 1;

    for (int i = 0; i < n; i++) {
        if (is_safe(board, i, col, n)) {
            board[i][col] = 1;
            if (solve(board, col + 1, n)) return 1;
            board[i][col] = 0;
        }
    }

    return 0;
}

int main() {
    int n = 8;
    int** board = malloc(n * sizeof(int*));
    for (int i = 0; i < n; i++) {
        board[i] = calloc(n, sizeof(int));
    }

    solve(board, 0, n);

    for (int i = 0; i < n; i++) {
        free(board[i]);
    }
    free(board);

    return 0;
}
